src/lv_objx/lv_cpicker.c

```diff
@@ -1279,83 +1303,40 @@ static lv_res_t lv_cpicker_signal(lv_obj_t * cpicker, lv_signal_t sign, void * p
     return res;
 }
 
-static void lv_cpicker_prev_hsv_save(lv_cpicker_ext_t * ext)
+static lv_res_t lv_cpicker_reset_hsv_if_double_clicked(lv_obj_t * cpicker,
+                                                       lv_cpicker_ext_t * ext)
 {
-    switch(ext->color_mode)
-    {
-    case LV_CPICKER_COLOR_MODE_HUE:
-        ext->prev_hue = ext->hue;
-        break;
-    case LV_CPICKER_COLOR_MODE_SATURATION:
-        ext->prev_saturation = ext->saturation;
-        break;
-    case LV_CPICKER_COLOR_MODE_VALUE:
-        ext->prev_value = ext->value;
-        break;
-    }
-}
-
-static void lv_cpicker_prev_hsv_restore(lv_obj_t * cpicker,
-                                        lv_cpicker_ext_t * ext)
-{
-    switch(ext->color_mode)
-    {
-    case LV_CPICKER_COLOR_MODE_HUE:
-        ext->prev_hue = ext->hue;
-        break;
-    case LV_CPICKER_COLOR_MODE_SATURATION:
-        ext->prev_saturation = ext->saturation;
-        break;
-    case LV_CPICKER_COLOR_MODE_VALUE:
-        ext->prev_value = ext->value;
-        break;
-    }
-    lv_cpicker_invalidate(cpicker, false);
-}
+    lv_res_t res;
 
-static void lv_cpicker_reset_hsv_if_double_clicked(lv_obj_t * cpicker,
-                                                   lv_cpicker_ext_t * ext)
-{
     if(lv_tick_elaps(ext->last_click) < 400)
     {
         switch(ext->color_mode)
         {
         case LV_CPICKER_COLOR_MODE_HUE:
-            ext->hue = LV_CPICKER_DEF_HUE;
-            ext->prev_hue = ext->hue;
+            ext->hsv.h = LV_CPICKER_DEF_HSV.h;
             break;
         case LV_CPICKER_COLOR_MODE_SATURATION:
-            ext->saturation = LV_CPICKER_DEF_SATURATION;
-            ext->prev_saturation = ext->saturation;
+            ext->hsv.s = LV_CPICKER_DEF_HSV.s;
             break;
         case LV_CPICKER_COLOR_MODE_VALUE:
-            ext->value = LV_CPICKER_DEF_VALUE;
-            ext->prev_value = ext->value;
+            ext->hsv.v = LV_CPICKER_DEF_HSV.v;
             break;
         }
+        ext->prev_hsv = ext->hsv;
         lv_cpicker_invalidate(cpicker, false);
+
+        res = lv_event_send(cpicker, LV_EVENT_VALUE_CHANGED, NULL);
```

Comment:
Technically, AFAIK, the color value hasn't really changed. The `LV_EVENT_VALUE_CHANGED` is really to convey changes in the value (for controls like sliders/switches).

Reply:
I can add a test to only emit if the value wasn't the default

Reply:
I was confused by your message so I decided to read through the whole function. Now I see that it is actually changing the color value. Sorry about that. Nothing needs to be changed here.